### ISSUE TITLE
results re-produced

### ISSUE THREAD

**Issue by russ-1234** [haiduo/DeepKD#1]
Hi authors, 

Thanks for sharing the code. 

I am currently trying to reproduce the results reported in your paper. I used the configuration provided in the released code and followed the described training procedure, but I have not been able to obtain results comparable to those reported in the paper. For example, the result of ResNet32x4-ResNet8x4 (w/o top-k) is around 76.3.

I would like to ask whether there are any additional hyperparameters or implementation details that are not included in the released configuration.

Thank you very much for your help.

**Comment by jiangchengsong** [comment]
> Hi authors,
> 
> Thanks for sharing the code.
> 
> I am currently trying to reproduce the results reported in your paper. I used the configuration provided in the released code and followed the described training procedure, but I have not been able to obtain results comparable to those reported in the paper. For example, the result of ResNet32x4-ResNet8x4 (w/o top-k) is around 76.3.
> 
> I would like to ask whether there are any additional hyperparameters or implementation details that are not included in the released configuration.
> 
> Thank you very much for your help.

<img width="216" height="102" alt="Image" src="https://github.com/user-attachments/assets/85015abf-4c6f-46ba-a2f4-128357b5c841" />

Hi,

Thank you for your interest in our work and for sharing your reproduction result.

Could you first clarify whether the result was obtained with **KD+Ours** or **DKD+Ours**? For ResNet32×4 → ResNet8×4 without top-k, the reported results are 76.69% for KD+Ours and 77.25% for DKD+Ours, so the comparison depends on the specific setting.

We did not use any additional or unpublished hyperparameters beyond the released configuration. However, differences in PyTorch/CUDA versions and GPU hardware may affect random-number generation and numerical behavior. Consequently, the model initialization and subsequent optimization trajectory may differ even when the same random seed is used.

In our experience, the result is relatively sensitive to the `DELTA_NCG` parameter. You may try setting it to `0.025` in the `DEEPKD` section, at the location highlighted in the figure below:

We suggest keeping the other settings unchanged initially so that the effect of `DELTA_NCG` can be evaluated independently.

If the discrepancy persists, please share whether you are using KD+Ours or DKD+Ours, together with your complete configuration, PyTorch/CUDA versions, GPU model, random seed, and training log. We will be happy to investigate further.

Best regards,  
The Authors

**Comment by russ-1234** [comment]
> > Hi authors,
> > Thanks for sharing the code.
> > I am currently trying to reproduce the results reported in your paper. I used the configuration provided in the released code and followed the described training procedure, but I have not been able to obtain results comparable to those reported in the paper. For example, the result of ResNet32x4-ResNet8x4 (w/o top-k) is around 76.3.
> > I would like to ask whether there are any additional hyperparameters or implementation details that are not included in the released configuration.
> > Thank you very much for your help.
> 
> <img alt="Image" width="216" height="102" src="https://private-user-images.githubusercontent.com/164507513/639450309-85015abf-4c6f-46ba-a2f4-128357b5c841.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODczMTA2NjAsIm5iZiI6MTc4NzMxMDM2MCwicGF0aCI6Ii8xNjQ1MDc1MTMvNjM5NDUwMzA5LTg1MDE1YWJmLTRjNmYtNDZiYS1hMmY0LTEyODM1N2I1Yzg0MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwODIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDgyMVQxMTA2MDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YjRmNzdkMjU0MjI3M2U4ZGMwMTgyZTBlYzY0NTMyZThkYWI4ZGQxNTkxMmQwYzc4ZTQwMjY3OWIyYjRjOTg1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ._f8ViMB-Enfk7-BbdByGyjEFEsFXtFmF-PoxYzH8fro">
> Hi,
> 
> Thank you for your interest in our work and for sharing your reproduction result.
> 
> Could you first clarify whether the result was obtained with **KD+Ours** or **DKD+Ours**? For ResNet32×4 → ResNet8×4 without top-k, the reported results are 76.69% for KD+Ours and 77.25% for DKD+Ours, so the comparison depends on the specific setting.
> 
> We did not use any additional or unpublished hyperparameters beyond the released configuration. However, differences in PyTorch/CUDA versions and GPU hardware may affect random-number generation and numerical behavior. Consequently, the model initialization and subsequent optimization trajectory may differ even when the same random seed is used.
> 
> In our experience, the result is relatively sensitive to the `DELTA_NCG` parameter. You may try setting it to `0.025` in the `DEEPKD` section, at the location highlighted in the figure below:
> 
> We suggest keeping the other settings unchanged initially so that the effect of `DELTA_NCG` can be evaluated independently.
> 
> If the discrepancy persists, please share whether you are using KD+Ours or DKD+Ours, together with your complete configuration, PyTorch/CUDA versions, GPU model, random seed, and training log. We will be happy to investigate further.
> 
> Best regards, The Authors

Thanks for your quick reply. 

The reported result was obtained using KD + DeepKD. I will try varying the value of DELTA_NCG while keeping the other settings unchanged. 

Thank you again for your clarification.

**Comment by jiangchengsong** [comment]
Hi,

Thank you for the clarification.

I have also attached the student_best checkpoint and the corresponding worklog.txt from our experiment for your reference. I hope they will be helpful for reproducing the result and comparing the training process.

[worklog.txt](https://github.com/user-attachments/files/31306490/worklog.txt)
https://drive.google.com/file/d/1qYnw3db5IXP7PjTY5RsT1szNlKeZ2ncr/view?usp=drive_link

Best regards,
The Authors
